doc/api/packages.md

```diff
@@ -17,8 +16,8 @@ initial input, or when referenced by `import` statements within ES module code:
 
 * Files ending in `.mjs`.
 
-* Files ending in `.js` when the nearest parent `package.json` file contains a
-  top-level field [`"type"`][] with a value of `"module"`.
+* Files ending in `.js` when the package boundary `package.json` file contains a
```

Comment:
I disagree with the term “package boundary.” That was discussed early on when the docs were originally written. The idea behind “scope” is that what we’ve created mirrors function scopes in JavaScript: one scope can be inside another. The “boundary” analogy lacks this concept. I think “boundary” makes the docs harder to understand, not easier.

Reply:
I think the problem is that we didn't create this idea. It already existed for things like babel configs - and has been called "package boundary" for years. https://github.com/babel/website/blame/master/docs/config-files.md

So, I think it makes sense to align with existing usage here.

Reply:
This was a specific change request in the previous PR from @isaacs that the term "scope" conflicts with npm scopes.

Reply:
> babel/website@`master`/docs/config-files.md (blame)

The only mention of "package boundary" I can find there is this:

> These cases will primarily cause issues for users with a monorepo structure, because if you have [example] the config will now be entirely ignored, because it is across a package boundary.

This usage of “package boundary” corresponds with how I understand the term: a border _between_ packages. It’s not a synonym for the contents of the package, or a drop-in replacement for “package scope.” If we use the phrase “package boundary” at all, it should be in this sense, the way Babel uses the phrase.

“Package scope” refers not to a border but to an _area_—the space contained within the borders/boundaries. This makes more sense for us because we’re describing things like “the `.js` files in the package scope.” You _could_ describe things between boundaries in such terms, like “the cities within the nation’s borders,” or you could be more direct with a phrase like “the cities in the nation.”

One thing we _can_ do, certainly, is minimize our usage of either phrase “package scope” or “package boundary.” The phrase “parent `package.json`“ is clearer than either “the controlling package scope `package.json`“ or “the controlling package boundary `package.json`.” The latter two both rely on jargon, a term that the reader might not know, that we ourselves need to define earlier in the document (for either “scope” or “boundary.”) I would be fine with a revision to our docs that finds other mentions of “package scope” and similarly rewrites them into clearer, non-jargon reliant alternatives. But we shouldn’t prefer “package boundary” over “package scope,” if we even need to mention package boundaries at all.

Reply:
> I would be fine with a revision to our docs that finds other mentions of “package scope” and similarly rewrites them into clearer, non-jargon reliant alternatives. But we shouldn’t prefer “package boundary” over “package scope,” if we even need to mention package boundaries at all.

@GeoffreyBooth this PR is exactly renaming all uses of "package scope" to instead refer to the "package boundary". There aren't any others in the docs.

Would it help if we ensure that we mention the boundary specifically instead of the area. For example we do this already in phrases like "the package.json file for the package boundary". We could ensure to use phrasing like that instead. Let me know if that might work for you.

Reply:
> this PR is exactly renaming all uses of “package scope” to instead refer to the “package boundary”

I know, this is the problem. The terms are not interchangeable, just as you can’t replace “nation” with “border.” They’re related, sure, but distinct.

If you want to address the concern about avoiding confusion with `npm` scope being a similar term, I think all we need to do is find all references to “package scope” and rewrite as many of them as possible to use descriptive language that avoids jargon (so neither “scope” nor “boundary” phrases). So sentences like “Files ending with `.mjs` are always loaded as ES modules regardless of package scope” could be written like “Files ending with `.mjs` are always loaded as ES modules regardless of the `"type"` field in the nearest parent `package.json`.” That’s clearer than using either “scope” or “boundary.”

We should probably keep the reference to package scope where it’s initially defined, and we could link all references to “nearest parent `package.json`“ and similar to the section where we introduce the package scope concept, since that section has the extra detail about how either `package.json` files or `node_modules` folders define the boundaries around scopes.

Reply:
@GeoffreyBooth the point is to deprecate the term package scope entirely. So we cannot then have a section on "package scopes". Are you proposing we do not replace it with anything new then? What shall we call this section instead?

Reply:
> the point is to deprecate the term package scope entirely

If that is your goal then I am strongly opposed.

Reply:
I have no goals here other than to get this merged so we can release this docs change.

In particular I am addressing [this comment](https://github.com/nodejs/node/pull/34748#discussion_r478566608) from @isaacs here.

In that comment he suggests using the term "package context". Would you be happier with that?

Alternatively we can also decide it is too late to change this terminology and simply stick with package scope certainly.

Reply:
Sorry, my last comment was too harsh, and then you replied before I could elaborate.

Do you mind reverting all the “package boundary” changes and I’ll take a crack at proposing suggestions to remove or minimize all mentions of “package scope”? Basically I want to keep wherever the text currently has “nearest parent `package.json`,” as I think that’s better than any of these alternatives, and I want to see if I can think of better phrasing for the few other mentions of “package scope.” My goal would be to rewrite it without just swapping in a replacement phrase that’s similarly unknown to readers.

Reply:
@GeoffreyBooth sure I've reverted all cases that previously referred to nearest parent and left just renames of "package scope" to "package boundary".